types/tablet_v2/wlr_tablet_v2_tool.c

```diff
@@ -366,6 +375,12 @@ void wlr_send_tablet_v2_tablet_tool_proximity_in(
 		return;
 	}
 
+	// Reinitialize the focus destroy events
+	wl_list_remove(&tool->surface_destroy.link);
+	wl_list_init(&tool->surface_destroy.link);
```

Comment:
No need to `wl_list_init` if `wl_signal_add` is called right after.

The reason we use `wl_list_init` is to be able to use `wl_list_remove` to unregister the listener unconditionally, without having to check whether we're listening to a signal or not.

Reply:
Makes sense, done :+1: